### PR TITLE
Configurable metadata fixes

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

Unified the feel and looks of multiple forms throughout the app. Dropped phone and address fields from Account Settings.

## Additional details

Things changed:
- Unified the looks and feel of multiple forms (fields ordering, placeholders, labels, etc.):
  1. REST Services Modal
  2. New Project Modal
  3. Project Settings
  4. New Library Item Modal
  5. Library Item Details Modal
  6. Modify Password
- Disabled unfinished tabs in Account Settings
- Removed the broken "X" close button from Account Settings
- Dropped `phone_number` and `address` fields from Account Settings
- Unified the base font size (introduced a variable)
- Unified the looks of TextBox and WrappedSelect (`react-select`) components, including error handling
- Some code cleanup included *(mea culpa)*

## Related issues

This is a followup to #3611
Part of #3554
Closes #3629

### CODE REVIEW

**Review by jnm — APPROVED** [review]
@magicznyleszek, I added a few extra commits and a few non-blocking comments

File: jsapp/js/components/modalForms/projectSettings.es6 [hunk 1]
```diff
@@ -125,6 +116,20 @@ class ProjectSettings extends React.Component {
...
+  getInitialFieldsFromAsset(asset) {
+    const fields = {};
+
+    fields.name = asset ? asset.name : '';
+    fields.description = asset?.settings ? asset?.settings.description : '';
+    fields.sector = asset?.settings ? asset?.settings.sector : null;
+    fields.country = asset?.settings ? asset?.settings.country : null;
+    fields['share-metadata'] = asset?.settings ? asset?.settings['share-metadata'] : false;
+    fields.operational_purpose = asset?.settings ? asset?.settings.operational_purpose : null;
+    fields.collects_pii = asset?.settings ? asset?.settings.collects_pii : null;
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/modalForms/projectSettings.es6]: Right, fixed :+1:

File: jsapp/js/components/modalForms/projectSettings.es6 [hunk 2]
```diff
@@ -191,45 +196,33 @@ class ProjectSettings extends React.Component {
...
-    if (typeof this.props.onProjectDetailsChange === 'function') {
-      this.props.onProjectDetailsChange({fieldName, fieldValue});
-    }
-  }
-
-  onNameChange(evt) {
-    this.setState({name: assetUtils.removeInvalidChars(evt.target.value)});
-    this.onAnyDataChange('name', assetUtils.removeInvalidChars(evt.target.value));
-  }
+  onAnyFieldChange(fieldName, newFieldValue) {
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/modalForms/projectSettings.es6:204]: I actually went a bit further: I renamed all 3 functions to `onAnyFieldChange` and changed all states to keep fields in `this.state.fields`. I also added a comment that these three components have a similar form handling.

File: jsapp/js/components/common/wrappedSelect.tsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ class WrappedSelect extends React.Component<WrappedSelectProps> {
             className={classNames.join(' ')}
             classNamePrefix='kobo-select'
             menuPlacement='auto'
+            placeholder={this.props.placeholder || t('Select…')}
```
Comment (jnm) [inline jsapp/js/components/common/wrappedSelect.tsx:37]: image

**Review by jnm — APPROVED** [review]

**Pull request merged**